src/backend/aspen/api/views/phylo_runs.py

```diff
@@ -182,3 +185,56 @@ async def kick_off_phylo_run(
     )
 
     return PhyloRunResponseSchema.from_orm(workflow)
+
+
+async def get_editable_phylorun_by_id(db, group_id, run_id, user):
+    aliased(PhyloRun)
+    query = (
+        sa.select(PhyloRun)
+        .filter(
+            sa.and_(
+                PhyloRun.group == user.group,  # This is an access control check!
+                PhyloRun.group_id
+                == group_id,  # This makes sure the group ID in our path is accurate
+                PhyloRun.id == run_id,
+            )
+        )
+        .options(joinedload(PhyloRun.outputs))
+    )
+    results = await db.execute(query)
+    try:
+        run = results.scalars().unique().one()
+    except NoResultFound:
+        raise ex.NotFoundException("phylo run not found")
+    if run.workflow_status == WorkflowStatusType.STARTED:
+        raise ex.NotFoundException("Can't modify an in-progress phylo run")
```

Comment:
nit: I think this is more of a `BadRequestException` since we were able to find the resource